uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/manage-leave-requests.component.js

```diff
@@ -254,36 +254,43 @@ define([
     }
 
     /**
-     * Loads the managees and calls loadLeaveRequests()
+     * Loads the leave requests for the given data type
      *
+     * @param {String} dataType - which data to load, table or filter or both
      * @return {Promise}
      */
-    function loadManageesAndLeaves () {
-      vm.loading.content = true;
+    function loadData (dataType) {
+      if (dataType) {
+        return loadLeaveRequests(dataType);
+      } else {
+        return $q.all([
+          loadLeaveRequests('table'),
+          loadLeaveRequests('filter')
+        ]);
+      }
+    }
 
+    /**
+     * Loads the managees and calls loadLeaveRequests()
+     *
+     * @param {String} dataType - which data to load, table or filter
+     * @return {Promise}
+     */
+    function loadManageesAndLeaves (dataType) {
```

Comment:
We do not always need to fetch contacts. But it was done to simplify the refresh logic, right now we have a single refresh function which always gets called whenever we want to do any kind of refresh. Normally its not a good idea because we will be calling the api unnecessarily, but in our case as the API is cached so we never call the same api twice for same filters. But it helps to simplify the code a lot.